pkgs/development/yottadb/default.nix

```diff
@@ -0,0 +1,198 @@
+{
+  lib,
+  stdenv,
+
+  binutils,
+  cmake,
+  coreutils-full,
+  curl,
+  elfutils,
+  fetchFromGitLab,
+  file,
+  gcc10,
+  git,
+  glibcLocales,
+  gnugrep,
+  gnumake,
+  gpgme,
+  icu,
+  libconfig,
+  libgcrypt,
+  libgpgerror,
+  locale,
+  makeWrapper,
+  ncurses,
+  openssl,
+  pkg-config,
+  tcsh,
+  util-linux,
+  writeText,
+  zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "yottadb";
+  version = "1.34";
+
+  src = fetchFromGitLab {
+    owner = "YottaDB";
+    repo = "DB/YDB";
+    rev = "r${version}";
+    leaveDotGit = true;
+    sha256 = "sha256-rYKw+R2VF+aht4UTcsiGVavIbmTRwpuUGHQe8LJCFQo=";
+  };
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [
+    file cmake gnumake curl tcsh
+    elfutils.dev libconfig libgcrypt libgpgerror
+    gpgme ncurses openssl zlib binutils
+    pkg-config icu.dev git glibcLocales
+    makeWrapper
+  ];
+
+  buildInputs = [
+    file elfutils.dev gcc10 ncurses binutils locale
+    pkg-config icu icu.dev glibcLocales
+    coreutils-full gnugrep
+    util-linux
+  ];
+
+  propagatedBuildInputs = [
+    file elfutils.dev ncurses binutils locale
+    pkg-config icu icu.dev glibcLocales
+    coreutils-full gnugrep
+    util-linux
+  ];
+
+  # The version number is generated using git-describe
+  # and any code patching makes this version number dirty.
+  # Do not modify the git repo in any way!
+  dontFixCmake = true;
+
+  preConfigure = ''
+    # The following cannot be passed to cmakeFlags directly due
+    # to variable substitution
+    cmakeFlags="$cmakeFlags -DYDB_INSTALL_DIR:STRING=$out/dist"
+  '';
+
+  preBuild = ''
+    # Make sure that we can find all the runtime libs in order to
+    # use the freshly built M compiler to build the M code...
+    export LD_LIBRARY_PATH="${lib.makeLibraryPath buildInputs}:$LD_LIBRARY_PATH"
+
+    # Revert any potential repo modifications to keep the release clean
+    git reset --hard
+  '';
+
+  # NOTE: Setting exec_prefix=@@YDB_EXEDIR@@, includedir=@@YDB_INCDIR@@
+  # and libdir=@@YDB_LIBDIR@@ breaks dependent software (YDBRust for
+  # instance) for some reason.
+  yottadbPkgConfigTemplate = writeText "yottadb.pc.in" ''
+prefix=@@YDB_DIST@@
```

Comment:
This can be indented normally, using spaces, because the string builder removes the spaces. I think. Maybe verify that…